third_party/proton/README.md

```diff
@@ -0,0 +1,117 @@
+# Proton - A Profiler for Triton
+
+## Introduction
+
+Proton is a lightweight profiler for Triton, designed to be used for code written in Python and to invoke underlying GPU kernels. Proton provides insightful information about the program context, metadata, and hardware performance metrics of the GPU kernels invoked.
+
+## Installation
+
+The following command installs the latest version of Proton.
+
+```bash
+git clone https://github.com/openai/triton
+cd triton/third_party/proton
+TRITON_BUILD_PROTON=1 pip install .
```

Comment:
Is there a reason not to build Proton by default?  Does it make the build significantly slower or something?

Reply:
Oh, this is because proton is only applied to nvidia GPUs currently and relies on cuda runtime. So I suppose it shouldn't be built on an AMD environment.

Reply:
Is it different from the other nvidia stuff we build but can't build when CUDA is unavailable?

Reply:
It's a bit tricky. In theory you could build with header files only.

However, you need to include 10+ header files from cuda runtime. I tried to avoid it by downloading `nvidia_cuda_cupti_cu12` and `nvidia_cuda_runtime_cu12` and use the header files there but doesn't seem to work.

Reply:
@malfet May I know what do you build kineto in an environment without cuda runtime?

Reply:
Like, IMO it would be nice if we didn't have to maintain multiple build configurations.  For example, are we going to have CI with and without Proton?

Downloading some extra files that you won't need on AMD feels to me like a small price to pay.  Or we could have a build flag that says "nvidia/AMD-only" and then test *those* on CI (but then are we going to test nvidia+AMD? i.e. we'd still have 3 configs)